pkg/operator/operator.go

```diff
@@ -294,14 +294,14 @@ func (c *vSphereProblemDetectorController) runChecks(ctx context.Context, cluste
 		// (i.e. retry as slow as allowed).
 		nextDelay = defaultBackoff.Cap
 	}
-	return nextDelay, nil
+	return nextDelay, checkError
 }
```

Comment:
Can we have a test that checks conditions after some of the checks fail? Just a unit test is fine.

Reply:
Unit test added